substrate/frame/election-provider-multi-phase/src/remote_mining.rs

```diff
@@ -0,0 +1,265 @@
+
+	pub struct MaxWeight;
+	impl Get<Weight> for MaxWeight {
+		fn get() -> Weight {
+			Weight::from_parts(
+				WEIGHT_REF_TIME_PER_SECOND.saturating_mul(2),
+				WEIGHT_PROOF_SIZE_PER_MB.saturating_mul(5),
+			)
+		}
+	}
+
+	generate_solution_type!(
+		#[compact]
+		pub struct PolkadotSolution::<
+			VoterIndex = u32,
+			TargetIndex = u16,
+			Accuracy = sp_runtime::PerU16,
+			MaxVoters = ConstU32<12_500>,
+		>(24)
+	);
+
+	/// Some configs are a bit inconsistent, but we don't care about them for now.
+	impl crate::MinerConfig for MinerConfig {
+		type AccountId = sp_runtime::AccountId32;
+		type MaxBackersPerWinner = ConstU32<1024>;
+		type MaxLength = ConstU32<{ 4 * 1024 * 1024 }>;
+		type MaxVotesPerVoter = ConstU32<24>;
+		type MaxWeight = MaxWeight;
+		type MaxWinners = ConstU32<1000>;
+		type Solution = PolkadotSolution;
+
+		fn solution_weight(
+			_voters: u32,
+			_targets: u32,
+			_active_voters: u32,
+			_degree: u32,
+		) -> Weight {
+			Default::default()
+		}
+	}
+}
+
+pub struct HackyGetSnapshot<T: MinerConfig>(PhantomData<T>);
+
+type UntypedSnapshotOf<T> = RoundSnapshot<
+	<T as MinerConfig>::AccountId,
+	frame_election_provider_support::Voter<
+		<T as MinerConfig>::AccountId,
+		<T as MinerConfig>::MaxVotesPerVoter,
+	>,
+>;
+
+impl<T: MinerConfig> HackyGetSnapshot<T> {
+	fn snapshot() -> UntypedSnapshotOf<T>
+	where
+		UntypedSnapshotOf<T>: Decode,
+	{
+		let key = [
+			sp_core::hashing::twox_128(b"ElectionProviderMultiPhase"),
+			sp_core::hashing::twox_128(b"Snapshot"),
+		]
+		.concat();
+		frame_support::storage::unhashed::get::<UntypedSnapshotOf<T>>(&key).unwrap()
+	}
+
+	fn desired_targets() -> u32 {
+		let key = [
+			sp_core::hashing::twox_128(b"ElectionProviderMultiPhase"),
+			sp_core::hashing::twox_128(b"DesiredTargets"),
+		]
+		.concat();
+		frame_support::storage::unhashed::get::<u32>(&key).unwrap()
+	}
+}
+
+pub type FakeBlock = sp_runtime::testing::Block<sp_runtime::testing::TestXt<(), ()>>;
+
+pub struct Balancing;
+impl Get<Option<BalancingConfig>> for Balancing {
+	fn get() -> Option<BalancingConfig> {
+		Some(BalancingConfig { iterations: 10, tolerance: 0 })
+	}
+}
+pub type SolverOf<T> = frame_election_provider_support::SequentialPhragmen<
+	<T as MinerConfig>::AccountId,
+	Perbill,
+	Balancing,
+>;
+
+fn test_for_network<T: MinerConfig>()
+where
+	UntypedSnapshotOf<T>: Decode,
+{
+	let snapshot = HackyGetSnapshot::<T>::snapshot();
+	let desired_targets = HackyGetSnapshot::<T>::desired_targets();
+
+	let (solution, score, _size, _trimming) =
+		Miner::<T>::mine_solution_with_snapshot::<SolverOf<T>>(
+			snapshot.voters.clone(),
+			snapshot.targets.clone(),
+			desired_targets,
+		)
+		.unwrap();
+
+	let raw_solution = RawSolution { round: 0, solution, score };
+
+	let _ready_solution = Miner::<T>::feasibility_check(
+		raw_solution,
+		ElectionCompute::Signed,
+		desired_targets,
+		snapshot,
+		0,
+		Default::default(),
+	)
+	.unwrap();
+}
+
+#[tokio::test]
+async fn mine_for_polkadot() {
```

Comment:
I have used this to test that the new check indeed doesn't cause anything to fail on a few recent staking elections on polkadot and kusama.